src/checks/consistency/single_greater_than_any.rs

```diff
@@ -0,0 +1,195 @@
+use crate::{ConsistencyCache, Flag, Timeseries, TimeseriesPair};
+
+/// Compares a single value to a higher resolution sequence, where the single value should never
+/// be greater than any value in the sequence (including an adjustment)
+///
+/// Returns:
+/// For the single value:
+/// - [`Flag::DataMissing`] if the single value is missing,
+/// - [`Flag::Fail`] if this invariant is broken (i.e the single value is greater than the any element
+///   of the sequence plus the adjustment),
+/// - [`Flag::DataMissing`] if any of the elements is missing, as we cannot be sure a missing data
+///   point did not violate the invariant.
+/// - [`Flag::Pass`] otherwise.
+///
+/// For each element in the sequence:
+/// - [`Flag::DataMissing`] if the single value or the element is missing,
+/// - [`Flag::Fail`] if this invariant is broken (i.e the single value is greater than the element
+///   plus the adjustment),
+/// - [`Flag::Pass`] otherwise.
```

Comment:
```suggestion
/// For the single value and for each element of the sequence:
/// - [`Flag::DataMissing`] if the single value or the element is missing,
/// - [`Flag::Fail`] if this invariant is broken (i.e the single value is greater than the element
///   plus the adjustment),
/// - [`Flag::Pass`] otherwise.
```
Can it be compressed?

Reply:
I think it's easier to reason about with them separate.

For one thing without them separate, it's not so obvious that the flags can be different. I also think it's not as clear under which conditions they are different and in what way.

Reply:
Yeah, it's probably better to be 100% explicit